pkg/ingress/rule_optimizer_test.go

```diff
@@ -229,6 +230,300 @@ func Test_defaultRuleOptimizer_Optimize(t *testing.T) {
+								},
+							},
+						},
+						Ingresspath: networking.HTTPIngressPath{
+							Path:     "/aaa/bbb",
+							PathType: (*networking.PathType)(awssdk.String("Prefix")),
+						},
+					},
+					{
+						Conditions: []elbv2model.RuleCondition{
+							{
+								Field: elbv2model.RuleConditionFieldPathPattern,
+								PathPatternConfig: &elbv2model.PathPatternConditionConfig{
+									Values: []string{"/*"},
+								},
+							},
+						},
+						Actions: []elbv2model.Action{
+							{
+								Type: elbv2model.ActionTypeRedirect,
+								RedirectConfig: &elbv2model.RedirectActionConfig{
+									Host:       awssdk.String("home.example.com"),
+									StatusCode: "HTTP_301",
+								},
+							},
+						},
+						Ingresspath: networking.HTTPIngressPath{
+							Path:     "/abc",
+							PathType: (*networking.PathType)(awssdk.String("Prefix")),
+						},
+					},
+				},
+			},
+			want: []Rule{
+				{
+					Conditions: []elbv2model.RuleCondition{
+						{
+							Field: elbv2model.RuleConditionFieldPathPattern,
+							PathPatternConfig: &elbv2model.PathPatternConditionConfig{
+								Values: []string{"/app"},
+							},
+						},
+					},
+					Actions: []elbv2model.Action{
+						{
+							Type: elbv2model.ActionTypeFixedResponse,
+							FixedResponseConfig: &elbv2model.FixedResponseActionConfig{
+								StatusCode: "200",
+							},
+						},
+					},
+					Ingresspath: networking.HTTPIngressPath{
+						Path:     "/aaa/bbb",
+						PathType: (*networking.PathType)(awssdk.String("Prefix")),
+					},
+				},
+				{
+					Conditions: []elbv2model.RuleCondition{
+						{
+							Field: elbv2model.RuleConditionFieldPathPattern,
+							PathPatternConfig: &elbv2model.PathPatternConditionConfig{
+								Values: []string{"/*"},
+							},
+						},
+					},
+					Actions: []elbv2model.Action{
+						{
+							Type: elbv2model.ActionTypeRedirect,
+							RedirectConfig: &elbv2model.RedirectActionConfig{
+								Host:       awssdk.String("home.example.com"),
+								StatusCode: "HTTP_301",
+							},
+						},
+					},
+					Ingresspath: networking.HTTPIngressPath{
+						Path:     "/abc",
+						PathType: (*networking.PathType)(awssdk.String("Prefix")),
+					},
+				},
+				{
+					Conditions: []elbv2model.RuleCondition{
+						{
+							Field: elbv2model.RuleConditionFieldPathPattern,
+							PathPatternConfig: &elbv2model.PathPatternConditionConfig{
+								Values: []string{"/www"},
+							},
+						},
+					},
+					Actions: []elbv2model.Action{
+						{
+							Type: elbv2model.ActionTypeRedirect,
+							RedirectConfig: &elbv2model.RedirectActionConfig{
+								Path:       awssdk.String("/app"),
+								StatusCode: "HTTP_301",
+							},
+						},
+					},
+					Ingresspath: networking.HTTPIngressPath{
+						Path:     "/",
+						PathType: (*networking.PathType)(awssdk.String("Exact")),
+					},
+				},
+			},
+		},
+		{
+			name: "Rules sorted by path length descending, when tie, rules with Exact path type precedes",
```

Comment:
also add a test with the same values for prefix and exact path, and verify the exact match takes precedence